### ISSUE TITLE
Question regarding generate positive pair sentences

### ISSUE THREAD

**Issue by cordeliazhu** [princeton-nlp/SimCSE#48]
Great works!  If I wanna using SimCSE in the doc query match scenario could this method work? The query contains more words than 'doc'.

**Comment by cordeliazhu** [comment]
Hi tianyu, 
after I read the trained train.py I found the code where you do the  'positive pair' embedding.

My question is what if in a sceneraio where I have pair datasets(s1,s2) could I still can generate 'positive pair'.? My understanding is that I need to change the following code to four columns [s1 , s1, s2 , s2]. Is that right? Looking forward for your reply.
if len(column_names) == 2:
        # Pair datasets
        sent0_cname = column_names[0]
        sent1_cname = column_names[1]
    elif len(column_names) == 3:
        # Pair datasets with hard negatives
        sent0_cname = column_names[0]
        sent1_cname = column_names[1]
        sent2_cname = column_names[2]
    elif len(column_names) == 1:
        # Unsupervised datasets
        sent0_cname = column_names[0]
        sent1_cname = column_names[0]
    else:
        raise NotImplementedError

**Comment by gaotianyu1350** [comment]
Hi,

You can definitely use this for doc-query match (though you may want to adjust the max sequence length for your documents might be composed of much more words. I didn't quite get your second question. If you use a pair dataset (which I think so fo r doc-query match), you should be able to directly use the `#pair datasets` part of data processing code.

**Comment by cordeliazhu** [comment]
Thanks tianyu! My understanding is that if I already have a pair datasets then simCSE will not be much helpful, for I do not need to generate similar representation. And since I do not have a hard negative sentence( in this paper z3) I can not use the loss  in equation 5. Is that right? Sorry to bother you again,

**Comment by gaotianyu1350** [comment]
You're right. However, you can also use similar loss functions for your pair dataset with dual-encoders (which is the case in many doc-query retrieval models).

**Comment by cordeliazhu** [comment]
Thanks!
